src/modules/navigator/rtl.cpp

```diff
@@ -56,6 +57,156 @@ RTL::on_inactive()
 {
 	// Reset RTL state.
 	_rtl_state = RTL_STATE_NONE;
+
+	find_RTL_destination();
+
+}
+
+void
+RTL::find_closest_landing_point()
+{
+	// get home position:
+	home_position_s &home_landing_position = *_navigator->get_home_position();
+	// get global position
+	const vehicle_global_position_s &global_position = *_navigator->get_global_position();
+
+	// set destination to home per default, then check if other valid landing spot is closer
+	_destination.set(home_landing_position);
+	// get distance to home position
+	double dlat = home_landing_position.lat - global_position.lat;
+	double dlon = home_landing_position.lon - global_position.lon;
+	double min_dist_squared = dlat * dlat + dlon * dlon;
+
+	if (_navigator->get_mission_start_land_available()) {
+		double mission_landing_lat = _navigator->get_mission_landing_lat();
+		double mission_landing_lon = _navigator->get_mission_landing_lon();
+
+		// compare home position to landing position to decide which is closer
+		dlat = mission_landing_lat - global_position.lat;
+		dlon = mission_landing_lon - global_position.lon;
+		double dist_squared = dlat * dlat + dlon * dlon;
+
+		if (dist_squared < min_dist_squared) {
+			min_dist_squared = dist_squared;
+			_destination.lat = _navigator->get_mission_landing_lat();
+			_destination.lon = _navigator->get_mission_landing_lon();
+			_destination.alt = _navigator->get_mission_landing_alt();
+			_destination_type = RTL_DESTINATION_MISSION_LANDING;
+
+		} else {
+			_destination_type = RTL_DESTINATION_HOME;
```

Comment:
Can you set this before `if (_navigator->get_mission_start_land_available()) {`?